### DISCUSSION TITLE
Requests time to time seem to stack and require more time to be served

### DISCUSSION THREAD

**Discussion by evilaliv3** [nextcloud/all-in-one#1300] [Questions] [answered]
I saw that many users reports that the web application is slow.

This is a known reality and the client on the typical infrastructure requires ~7s to load but here i'm reporting something different.

While using AIO on a new setup, i experience that time to time some requests stack and take more time to execute.

The behavior has been identified even in situation of no load.
E.g. during simple tests, making a pause and then refreshing the client the first request could pass from executing in few seconds to take 20-45 seconds.
During this period apparently the server seems to not incur in any heavy log, and the main active process (representing %4 of the load) is Redis.

**Comment by szaimen** [comment]
Hi, 

can you please be much more verbose with your report? I have the impression that it is missing a lot of information that is needed for finding solutions against your problem!

Thank you!

**Comment by evilaliv3** [comment]
Thank you Simon

Actually I'm trying to debug the situation but at the moment Ive really no
clue.

If you can please let me know how would you approach to the debug of this
situation

**Comment by szaimen** [comment]
Lets move this to discussions first and think about debug steps then 👍

**Comment by szaimen** [comment]
First, can you please send the command that you used to create the mastercontainer? Also is aio running behind a reverse proxy?

**Comment by evilaliv3** [comment]
Thank you @szaimen 

First of all the setup features a reverse proxy configured up to you recommendations.

The command used to create the master container is:

sudo docker run --sig-proxy=false --name nextcloud-aio-mastercontainer --restart always --publish 8080:8080 -e APACHE_PORT=11000 --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest

Here you could see two screenshots about the loading of the application.
The first screenshoot feature a fast load of 5 seconds, the second a slow load of 26 seconds.
The difference is the answer by the server to the first request for the dashboard.

![Screenshot from 2022-10-24 11-47-39](https://user-images.githubusercontent.com/217034/197499985-9f477cc3-2735-4a0b-a9eb-8aba187e432f.png)

![Screenshot from 2022-10-24 11-49-06](https://user-images.githubusercontent.com/217034/197500044-75baaa87-1adc-4326-9b57-155869f39d36.png)


I notice that is more probably to hit a slow loading if i wait some time before accessing again the application (maybe redis is configured with a very short caching period? this is a very fast guess not supported by any evidence at the moment).

**Comment by evilaliv3** [comment]
I attach as well the specs of the system.

The VM that hosts AIO features an  Intel(R) Xeon(R) CPU E5-2630 v3 @ 2.40GHz (8 cores) and 8GB of RAM.

![Screenshot from 2022-10-24 12-00-52](https://user-images.githubusercontent.com/217034/197501543-4f631fb1-ce43-4be4-8c0d-bd257ff431b0.png)

**Comment by evilaliv3** [comment]
@szaimen : could this be related to the feature of antibruteforce?

Apparently the request that stuck is just the first an take ~25-50 seconds that is very near to the value declared for the anti bruteforce mechanism: https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/bruteforce_configuration.html

I would like to test disabling this. Do you know how could i and how to override this permanently in AIO ?

**Reply by szaimen** [reply]
That is an interesting idea! 

See https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/occ_command.html#security

and this app:  https://apps.nextcloud.com/apps/bruteforcesettings

**Comment by evilaliv3** [comment]
Thank you, i saw this general guide but i wonder how to issue this on AIO

**Reply by szaimen** [reply]
Maybe this helps? https://github.com/nextcloud/all-in-one#bruteforce-protection-faq

**Comment by szaimen** [comment]
@evilaliv3 do you see the correct ip-addresses of remote devices in the Nextcloud logs (e.g. when a login attempt is incorrect)? 

Otherwise you might not set the headers correctly which might trigger the bruteforce-protection in the long run...

**Comment by evilaliv3** [comment]
This helps removing a block but my guess is that the anti brutefore mechanism may be impacting active sessions.

I will try to put an exception for every IP and see the result

**Reply by szaimen** [reply]
better would be fixing your reverse proxy config so that the ip-addresses actually land in the Nextcloud logs which is used for brute-force protection...

**Comment by evilaliv3** [comment]
@szaimen : is there a way i could look into the apache log? i would like to see the IP that they have received till now

**Reply by szaimen** [reply]
Not really. which ip-address lands in the Nextcloud logs when a login with a false password is happening?

**Comment by evilaliv3** [comment]
bingo: docker exec 4cffa8ee0a12 cat /var/log/apache2/access.log

**Comment by evilaliv3** [comment]
the ip that is received is 127.0.0.1

**Reply by szaimen** [reply]
So in the nextcloud logs so e.g. `https://yourdomain.com/settings/admin/logging` is `127.0.0.1` logged when someone fails to log in?

**Comment by evilaliv3** [comment]
This is the case! someone is probably trying to bruteforce the GlobaLekas nextcloud as we speak!

![Screenshot 2022-10-24 at 20-00-14 Settings - GlobaLeaks Cloud](https://user-images.githubusercontent.com/217034/197594294-baacbd90-254e-4ed1-8540-7b6d678fcc79.png)

**Comment by szaimen** [comment]
So as I said, you need to fix your reverse proxy.

In Nginx, this would be:

```
        proxy_set_header Host $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
```

**Comment by evilaliv3** [comment]
Nope, actually nextcloud internally receives the correct ip and i see that the attacker is using the static ip: 93.45.84.12

I wonder if the anti bruteforcing mechanism could be using the proxy ip instead of the forwarded for.

**Reply by szaimen** [reply]
> I wonder if the anti bruteforcing mechanism could be using the proxy ip instead of the forwarded for. 

in theory it should not as trusted_proxies contains `127.0.0.1` IIRC...

**Comment by evilaliv3** [comment]
i correct myself, 93.45.84.12 is my current IP assigned in the dynamic assignment of my provider.

This means that as i was guessing there is something bad in nextcloud that recognize a reload as a new attempt of login. I wonder how this could be possible.

**Comment by evilaliv3** [comment]
It  seems to be logging an attempt of login every minute and exactly every minute i see in my client the request to the /rooms API ...

**Comment by evilaliv3** [comment]
Other users report this:
https://help.nextcloud.com/t/nextcloud-with-reverse-proxy/147838
https://dietpi.com/forum/t/nextcloud-to-slow/14625

I guess this could be a bug in nextcloud, do you know in the team to who we shall notify this?

**Comment by szaimen** [comment, marked as answer]
Looks like https://github.com/nextcloud/server/issues/33919 then

**Comment by evilaliv3** [comment]
absolutely

**Comment by 4tler** [comment]
i make the same observation as evilaliv3.
I sometimes had the situation that Talk, for example, was no longer usable for several minutes. If I understand the statements in the bug marked as a solution correctly, it seems to be an old problem that has only recently appeared in the LOG. However, I recently moved from a direct LXC configuration to AIO and had no problems before. NC ran much faster in LXC. 

The newly created wiki ( #1335 ) mean that an installation in a VM is not recommended. I have everything running in an LXC and unfortunately suspect that the rather sluggish performance has something to do with this.

Is there a possibility to start a poll within Github to find out if other users with VM, LXC or native installation also have performance problems?

Or can the high loading time really be due to the bug masked as a solution here?

**Comment by szaimen** [comment]
Due to bad login attempts from his ip-address seems the brute-force protection to be triggered which explains the behaviour in @evilaliv3's case.

**Reply by 4tler** [reply]
ok, that means that if I have e.g. 10 users, they all trigger the error in short intervals because they access NC with the same ip (internal network). Makes sense. This also explains that the error usually occurs when the day is somewhat advanced and several users are working with it. Then the fix should solve the problem after all

This has helped us a lot. thank you very much and especially thank you for this great project.

**Comment by evilaliv3** [comment]
@szaimen is there any update on this? actually from my side im still getting this error even if i'm alone testing my system.

It seems related to old sessions expiring on some tabs and starting causing failed logged request.
If i'm not wrong other users were reporting thi.

**Reply by szaimen** [reply]
Sorry, I cannot resolve this myself. It is something upstream needs to fix. 

A possible workaround is that you simply disable the brute-force protection for specific ip-addresses that you trust.
